hi-ml/src/health_ml/run_ml.py

```diff
@@ -0,0 +1,233 @@
+                 project_root: Optional[Path] = None) -> None:
+        """
+        Driver class to run a ML experiment. Note that the project root argument MUST be supplied when using hi-ml
+        as a package!
+
+        :param container: The LightningContainer object to use for training.
+        :param project_root: Project root. This should only be omitted if calling run_ml from the test suite. Supplying
+        it is crucial when using hi-ml as a package or submodule!
+        """
+        self.container = container
+        self.experiment_config = experiment_config
+        self.container.num_nodes = self.experiment_config.num_nodes
+        self.project_root: Path = project_root or fixed_paths.repository_root_directory()
+        self.storing_logger: Optional[StoringLogger] = None
+        self._has_setup_run = False
+
+    def setup(self, azure_run_info: Optional[AzureRunInfo] = None) -> None:
+        """
+        If the present object is using one of the built-in models, create a (fake) container for it
+        and call the setup method. It sets the random seeds, and then creates the actual Lightning modules.
+
+        :param azure_run_info: When running in AzureML or on a local VM, this contains the paths to the datasets.
+        This can be missing when running in unit tests, where the local dataset paths are already populated.
+        """
+        if self._has_setup_run:
+            return
+        if azure_run_info:
+            # Set up the paths to the datasets. azure_run_info already has all necessary information, using either
+            # the provided local datasets for VM runs, or the AzureML mount points when running in AML.
+            # This must happen before container setup because that could already read datasets.
+            if len(azure_run_info.input_datasets) > 0:
+                input_datasets = azure_run_info.input_datasets
+                assert len(input_datasets) > 0
+                local_datasets = [
+                    check_dataset_folder_exists(input_dataset) for input_dataset in input_datasets  # type: ignore
+                ]
+                self.container.local_datasets = local_datasets  # type: ignore
+        # Ensure that we use fixed seeds before initializing the PyTorch models
+        seed_everything(self.container.get_effective_random_seed())
+
+        # Creating the folder structure must happen before the LightningModule is created, because the output
+        # parameters of the container will be copied into the module.
+        self.container.create_filesystem(self.project_root)
+
+        self.container.setup()
+        self.container.create_lightning_module_and_store()
+        self._has_setup_run = True
+
+    @property
+    def is_offline_run(self) -> bool:
+        """
+        Returns True if the present run is outside of AzureML, and False if it is inside of AzureML.
+
+        :return:
+        """
+        return not is_running_in_azure_ml(RUN_CONTEXT)
```

Comment:
we have the negation of that already on container level, do we really need that?